.github/workflows/clang-format.yml

```diff
@@ -0,0 +1,34 @@
+name: clang-format-check
+
+on:
+  pull_request:
+    branches:
+    - sycl
+    types: [open, edit, reopen, synchronize]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        # checkout PR head
+        ref: '${{github.event.pull_request.head.sha}}'
+    - name: Fetch target branch
+      run: git fetch --no-tags --prune --depth=1 origin +refs/heads/${{github.base_ref}}:refs/remotes/origin/${{github.base_ref}}
+
+    - name: Get clang-format first
+      run: sudo apt-get install clang-format-9
+
+    - name: Run clang-format for the patch
+      run: |
+        git diff -U0 --no-color origin/${{github.base_ref}}...HEAD | ./clang/tools/clang-format/clang-format-diff.py -p1 -binary clang-format-9 > ./clang-format.patch
```

Comment:
I think something like

```
git clang-format --diff -q sycl > clang-format.patch
```

would do the same.

Reply:
Yes. And this is not what I wanted. :-)
While I was replying to this comment I found a bug.
The command you suggesting will check formatting for the pull request changes + all the changes done in `sycl` branch after PR branch point. 
If someone else commits unformatted code (e.g. Vladimir might merge community changes) I don't want my pre-commit check to fail.
Does git-clang-format script support commit ranges?
Also it's not standard git command, so we also have to add path to the script to `PATH` environment variable.

Reply:
> Does git-clang-format script support commit ranges?

Yes, but the syntax seem to be non-standard:

https://github.com/intel/llvm/blob/c494112b40f00de974d537a593ca162dcc011ede/clang/tools/clang-format/git-clang-format#L38-L43

> Also it's not standard git command, so we also have to add path to the script to PATH environment variable.

`apt-get` takes care of this.

Reply:
Are you saying that clang-format-9 package installs `git-clang-format` script?

From the description above I don't see how to check only PR commits, but not in the target branch. Are you suggesting something like
```bash
git clang-format --diff -q `git merge-base origin/${{github.base_ref}} HEAD` HEAD > clang-format.patch
```
?

Reply:
> Are you saying that clang-format-9 package installs git-clang-format script?

Yes, I checked it a moment ago.

> Are you suggesting something like?

No, just:

```
git clang-format --diff -q origin/${{github.base_ref}} HEAD > clang-format.patch
```

Since PR is just a branch isn't it enough to check the diff between HEAD and the last commit before PR change list?

Anyway, I'm ok with your command line.

Reply:
> _UPD_: I think I get what you're talking about. If there are merge commits in the PR, they'll be ckecked, too. And clang-format can't handle it on its own.

Not only that: `github.base_ref` == `sycl` i.e. it's target branch for PR, so if `sycl` branch has commits, which are not in the PR branch, they will be checked as well.